pkg/cloudprovider/aws.go

```diff
@@ -0,0 +1,107 @@
+package cloudprovider
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+	"github.com/sirupsen/logrus"
+
+	"github.com/libopenstorage/cloudops"
+	awsops "github.com/libopenstorage/cloudops/aws"
+)
+
+type aws struct {
+	defaultProvider
+
+	ops cloudops.Ops
+}
+
+func (a *aws) init() error {
+	// Do parent init first
+	if err := a.defaultProvider.init(); err != nil {
+		return err
+	}
+	if a.distributionName != eksDistribution {
+		return nil
+	}
+
+	// Initialize aws client for cloud drive permission checks
+	client, err := awsops.NewClient()
+	if err != nil {
+		return err
+	}
+	a.ops = client
+	return nil
+}
+
+func (a *aws) Name() string {
+	return string(cloudops.AWS)
+}
+
+func (a *aws) CheckCloudDrivePermission() error {
+	// check the permission here by doing dummy drive operations
+	logrus.Infof("JLIAO: doing aws permisison check")
+	volTypeGp2 := opsworks.VolumeTypeGp2
+	volID := "px-check-aws-cloud-permission"
+	volSize := int64(1)
+
+	// Create
+	v, err := a.ops.Create(&ec2.Volume{
+		VolumeId:   &volID,
+		Size:       &volSize,
+		VolumeType: &volTypeGp2,
+	}, nil)
+	if err != nil {
+		logrus.Errorf("JLIAO: failed to create aws volume: %v", err)
+		return err
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"volume": v,
+	}).Infof("JLIAO: volume created")
+
+	// Attach
+	vol := v.(*ec2.Volume)
+	path, err := a.ops.Attach(*vol.VolumeId, nil)
+	if err != nil {
+		logrus.Errorf("JLIAO: failed to attach aws volume: %v", err)
+		return err
+	}
+	logrus.WithFields(logrus.Fields{
+		"path": path,
+	}).Infof("JLIAO: volume created")
+
+	// Tag
+	if err := a.ops.ApplyTags(volID, map[string]string{"testkey": "testval"}); err != nil {
+		logrus.Errorf("JLIAO: failed to tag %v", err)
+		return err
+	}
+
+	// Expand
+	if _, err := a.ops.Expand(volID, 2); err != nil {
+		logrus.Errorf("JLIAO: failed to expand %v", err)
+		return err
+	}
```

Comment:
Let's skip this one for now. It will take a long time to finish the preflight and the expand won't block the initial installation either.